git-filter-branch.sh

```diff
@@ -83,15 +83,21 @@ set_ident () {
 	finish_ident COMMITTER
 }
 
-if test -z "$FILTER_BRANCH_SQUELCH_WARNING$GIT_TEST_DISALLOW_ABBREVIATED_OPTIONS"
+if test -z "$FILTER_BRANCH_SQUELCH_WARNING$GIT_TEST_DISALLOW_ABBREVIATED_OPTIONS" && \
+   test "$(git config --bool filter-branch.squelchWarning)" != true
 then
 	cat <<EOF
 WARNING: git-filter-branch has a glut of gotchas generating mangled history
-	 rewrites.  Hit Ctrl-C before proceeding to abort, then use an
+	 rewrites. Hit Ctrl-C before proceeding to abort, then use an
 	 alternative filtering tool such as 'git filter-repo'
-	 (https://github.com/newren/git-filter-repo/) instead.  See the
+	 (https://github.com/newren/git-filter-repo/) instead. See the
 	 filter-branch manual page for more details; to squelch this warning,
-	 set FILTER_BRANCH_SQUELCH_WARNING=1.
+         set FILTER_BRANCH_SQUELCH_WARNING=1 or run the following command:
+
+           git config filter-branch.squelchWarning true
+         
+         You can replace "git config" with "git config --global" to disable
+         the warning for all repositories.
```

Comment:
Please use tabs instead of spaces to indent the lines.

Reply:
Fixed in latest commit bef1de4.